api/internal/localizer/localizer.go

```diff
@@ -127,3 +135,66 @@ func (lc *Localizer) localizeFile(path string) (string, error) {
 	}
 	return locPath, nil
 }
+
+// localizeBuiltinPlugins localizes built-in plugins on kust that can contain file paths. The built-in plugins
+// can be inline or in a file. This excludes helm.
+//
+// Note that the localization in this function has not been implemented yet.
+func (lc *Localizer) localizeBuiltinPlugins(kust *types.Kustomization) error {
+	for fieldName, plugins := range map[string]struct {
+		entries   []string
+		localizer kio.Filter
+	}{
+		"generators": {
+			kust.Generators,
+			&localizeBuiltinGenerators{},
+		},
+		"transformers": {
+			kust.Transformers,
+			&localizeBuiltinTransformers{},
+		},
+	} {
+		for i, entry := range plugins.entries {
+			rm, isPath, err := lc.loadResource(entry)
+			if err != nil {
+				return errors.WrapPrefixf(err, "unable to load %s entry", fieldName)
+			}
+			err = rm.ApplyFilter(plugins.localizer)
+			if err != nil {
+				return errors.Wrap(err)
+			}
+			localizedPlugin, err := rm.AsYaml()
+			if err != nil {
+				return errors.WrapPrefixf(err, "unable to serialize localized %s entry %q", fieldName, entry)
+			}
+			var newEntry string
+			if isPath {
+				// TODO(annasong): write localizedPlugin to dst
+				newEntry = entry
+			} else {
+				newEntry = string(localizedPlugin)
+			}
+			plugins.entries[i] = newEntry
+		}
+	}
+	if len(kust.Validators) > 0 {
+		return errors.Errorf("no built-in validators with file paths, but validators field non-empty")
+	}
+	return nil
+}
+
+// loadResource tries to load resourceEntry as a file path or inline.
+// On success, loadResource returns the loaded resource map and whether resourceEntry is a file path.
+func (lc *Localizer) loadResource(resourceEntry string) (resmap.ResMap, bool, error) {
+	var fileErr error
+	rm, inlineErr := lc.rFactory.NewResMapFromBytes([]byte(resourceEntry))
+	if inlineErr != nil {
+		rm, fileErr = lc.rFactory.FromFile(lc.ldr, resourceEntry)
+		if fileErr != nil {
+			return nil, false, errors.WrapPrefixf(fileErr, `unable to load resource entry %q:
+when parsing as inline received error: %s
+when parsing as filepath received error`, resourceEntry, inlineErr)
```

Comment:
Does %s print out the error correctly? With errors we typically do %w

Reply:
Yes, when the argument of "%s" is an `error`, the `Error()` method is called and printed. 

We can only use `%w` with `fmt.Errorf`. However, if we use `fmt.Errorf`, we'd only be able to use `%w` once, but we have 2 errors that we're nesting here, so we'd need to use `%s` for 1 of them. This is also why I proposed creating a new error: https://github.com/kubernetes-sigs/kustomize/pull/4869#discussion_r1026957899, so that we can "unwrap" both errors in the face of the single `%w` constraint. 

Since we want to check for errors anyway (https://github.com/kubernetes-sigs/kustomize/pull/4869#discussion_r1033752986), I'll create a new error for now and I'll make changes based on your feedback.